docs/compilers/CSharp/Runtime Async Design.md

```diff
@@ -136,11 +140,38 @@ for given scenarios are elaborated in more detail below.
 
 TODO: Async iterators (returning `IAsyncEnumerable<T>`)
 
-#### `Task`, `Task<T>`, `ValueTask`, `ValueTask<T>` Scenarios
-
-For any lvalue of one of these types, we'll generally rewrite `await expr` into `System.Runtime.CompilerServices.RuntimeHelpers.Await(expr)`. A number of different example scenarios for this are covered below. The
+#### `RuntimeHelpers.Await` Scenarios
+
+The for any `await expr` with type `E`, the compiler will attempt to match it to a helper method in `System.Runtime.CompilerServices.RuntimeHelpers`. The following algorithm is used:
```

Comment:
>The for [](http://example.com/codeflow?start=0&length=7)

Extraneous "the"?

Reply:
nit: the sentence suggests that `await expr` has type `E`, but you mean that `expr` has type `E`.